MetaMask/metamask-extension
# Adding `TextField` component

## Explanation

Adding `TextField` component to allow for inputing and editing text

`TextField` wraps `TextFieldBase` and only supports controlled inputs as well as a `showClear` opton which displays a X clear button to allow for clearing text.

## More Information

* Fixes #15101 
* Based on https://github.com/MetaMask/metamask-extension/pull/16043
* Will automatically inherit updates from https://github.com/MetaMask/metamask-extension/pull/16201

## Screenshots/Screencaps

https://user-images.githubusercontent.com/8112138/195807335-62d1c31e-f536-4f15-9b6b-33d8980ea934.mov

## Manual Testing Steps

- Go to the latest storybook build on this PR
- Search TextField in search field (make sure it's the new one in component-library)
- Check controls work
- Check stories
- Check docs
- Check console

## Pre-Merge Checklist


### + If there are functional changes:
